packages/dashboard-frontend/src/services/oauth/index.ts

```diff
@@ -27,22 +27,16 @@ export class OAuthService {
     }
   }
 
-  static async refreshTokenIfNeeded(workspace: devfileApi.DevWorkspace): Promise<void> {
-    // if workspace is not created yet, do not refresh token
-    if (!workspace.status || !workspace.status.mainUrl) {
-      return;
-    }
-    if (!workspace.spec.template.projects) {
-      return;
-    }
-
-    const project = workspace.spec.template.projects[0];
-    if (!project || !project.git) {
-      return;
-    }
+  static async refreshTokenIfProjectExists(workspace: devfileApi.DevWorkspace): Promise<void> {
```

Comment:
@tolusha I'm a bit lost, why project presence in devfile is indicator for token refresh ?

Reply:
at the very least, would be great to have some comment with method description

Reply:
We need a repository url to figure out the correct OAuth provider.